python/lsst/meas/algorithms/defects.py

```diff
@@ -53,3 +72,350 @@ def policyToBadRegionList(policyFile):
+        r = schema.addField("R", type="ArrayD", size=2, units="pix")
+        rotang = schema.addField("ROTANG", type="D", units="deg")
+        component = schema.addField("COMPONENT", type="I")
+        table = lsst.afw.table.BaseCatalog(schema)
+
+        for i, defect in enumerate(self._defects):
+            box = defect.getBBox()
+            record = table.addNew()
+            record.set(x, box.getCenterX())
+            record.set(y, box.getCenterY())
+            record.set(shape, "BOX")
+            record.set(r, np.array([box.getWidth(), box.getHeight()], dtype=np.float64))
+            record.set(rotang, 0.0)
+            record.set(component, i)
+
+        # Set some metadata in the table (force OBSTYPE to exist)
+        metadata = copy.copy(self.getMetadata())
+        metadata["OBSTYPE"] = self._OBSTYPE
+        table.setMetadata(metadata)
+
+        return table
+
+    def writeFits(self, *args):
+        """Write defect list to FITS.
+
+        Parameters
+        ----------
+        *args
+            Arguments to be forwarded to
+            `lsst.afw.table.BaseCatalog.writeFits`.
+        """
+        table = self.toTable()
+
+        # Add some additional headers useful for tracking purposes
+        metadata = table.getMetadata()
+        now = datetime.datetime.utcnow()
+        metadata["DATE"] = now.isoformat()
+        metadata["CALIB_CREATION_DATE"] = now.strftime("%Y-%m-%d")
+        metadata["CALIB_CREATION_TIME"] = now.strftime("%T %Z").strip()
+
+        table.writeFits(*args)
+
+    @classmethod
+    def fromTable(cls, table):
+        """Construct a `Defects` from the contents of a
+        `~lsst.afw.table.BaseCatalog`.
+
+        Parameters
+        ----------
+        table : `lsst.afw.table.BaseCatalog`
+            Table with one row per defect.
+
+        Returns
+        -------
+        defects : `Defects`
+            A `Defects` list.
+        """
+
+        defectList = []
+
+        schema = table.getSchema()
+
+        # Check schema to see which definitions we have
+        if "X" in schema and "Y" in schema and "R" in schema and "SHAPE" in schema:
+            # This is a FITS region style table
+            isFitsRegion = True
+
+        elif "x0" in schema and "y0" in schema and "width" in schema and "height" in schema:
+            # This is a classic LSST-style defect table
+            isFitsRegion = False
+
+        else:
+            raise ValueError("Unsupported schema for defects extraction")
+
+        for r in table:
+            record = r.extract("*")
+
+            if isFitsRegion:
+                if record["SHAPE"] != "BOX":
+                    log.warning("Defect lists can only be defined using BOX not %s",
+                                record["SHAPE"])
+                box = lsst.geom.Box2I.makeCenteredBox(lsst.geom.Point2D(record["X"], record["Y"]),
+                                                      lsst.geom.Extent2I(record["R"]))
```

Comment:
Am I correct that this will always make the minimal box that contains the defect if it is not "BOX"?

Reply:
No. I've actually just fixed that logic. After the warning there was meant to be a `continue` so that it wouldn't break.  I did add support for POINT whilst I was there.